mern-ecommerce/controllers/auth.js

```diff
@@ -62,20 +62,20 @@ exports.requireSignin = expressJwt({
   userProperty: 'auth',
 });
 
-exports.isAuth = (req, res, next) => {
-  let user = req.profile && req.auth && req.profile._id == req.auth._id;
-  if (!user) {
+exports.isAdmin = (req, res, next) => {
+  if (req.profile.role === 0) {
     return res.status(403).json({
-      error: 'Access denied',
+      error: 'Admin resource! Access denied',
     });
   }
   next();
 };
 
-exports.isAdmin = (req, res, next) => {
-  if (req.profile.role === 0) {
+exports.isAuth = (req, res, next) => {
+  let user = req.profile && req.auth && req.profile._id == req.auth._id;
```

Comment:
The condition `let user = req.profile && req.auth && req.profile._id == req.auth._id;` is not null-safe. It will throw an error if `req.profile` or `req.auth` is null or undefined. Consider using optional chaining (`?.`) or a null check before accessing properties.